Nos/Service/GiftWrapper.swift

```diff
@@ -0,0 +1,180 @@
+import Foundation
+import NostrSDK
+
+enum GiftWrapperError: Error {
+    case invalidPrivateKey
+    case invalidPublicKey
+    case serializationError
+    case signedRumorError
+}
+
+let TWODAYS: TimeInterval = 2 * 24 * 60 * 60
+struct NIP44v2Encrypter: NIP44v2Encrypting {}
+
+enum GiftWrapper: NIP44v2Encrypting {
+    /// Encrypts and gift-wraps a direct message (See NIP-17, NIP-44 and NIP-59).
+    static func wrap(
+        rumor: JSONEvent,
+        senderKeyPair: KeyPair,
+        receiverPubkey: RawAuthorID
+    ) throws -> JSONEvent {
+        let validatedRumor = try validateRumor(rumor)
+
+        let encryptedRumor = try encryptJSONEvent(
+            validatedRumor,
+            senderKeyPair: senderKeyPair,
+            receiverPubkey: receiverPubkey
+        )
+        
+        let sealCreatedAt = randomTimeUpTo2DaysInThePast()
+        var seal = JSONEvent(
+            pubKey: senderKeyPair.publicKeyHex,
+            createdAt: sealCreatedAt,
+            kind: .seal,
+            tags: [],
+            content: encryptedRumor
+        )
+        try seal.sign(withKey: senderKeyPair)
+
+        guard let randomKeyPair = KeyPair() else {
+            throw GiftWrapperError.invalidPrivateKey
+        }
+
+        let encryptedSeal = try encryptJSONEvent(
+            seal,
+            senderKeyPair: randomKeyPair,
+            receiverPubkey: receiverPubkey
+        )
+        
+        let wrapCreatedAt = randomTimeUpTo2DaysInThePast()
+        var giftWrap = JSONEvent(
+            pubKey: randomKeyPair.publicKeyHex,
+            createdAt: wrapCreatedAt,
+            kind: .giftWrap,
+            tags: [["p", receiverPubkey]],
+            content: encryptedSeal
+        )
+        
+        try giftWrap.sign(withKey: randomKeyPair)
+        
+        // Unsigned, publication will do that
```

Comment:
I think it is signed, though? I believe the `sign` function mutates the struct in place. And I think we do need to return it signed because it is signed with the random keypair, whereas the `RelayService.publish..` function will use the current user's keypair. Maybe we need to add a `RelayService.publishSigned` function, or maybe we already have one?

Reply:
Added this in https://github.com/planetary-social/nos/pull/1048/commits/e86b0ac4f32fd7e1e17dcd20bf0638714983bedc